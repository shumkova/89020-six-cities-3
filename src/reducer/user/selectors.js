import NameSpace from "../name-space.js";

const NAME_SPACE = NameSpace.USER;


const getAuthorizationStatus = (state) => {
  return state[NAME_SPACE].authorizationStatus;
};

const getAuthInfo = (state) => {
  return state[NAME_SPACE].authInfo;
};

export {getAuthorizationStatus, getAuthInfo};

