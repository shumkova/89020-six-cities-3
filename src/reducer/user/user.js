import {extend} from "../../utils";
import EditAuthInfo from "../../adapters/edit-auth-info";
import {AppRoute} from "../../const";
import history from "../../history";

const AuthorizationStatus = {
  AUTH: `AUTH`,
  NO_AUTH: `NO_AUTH`,
};

const initialState = {
  authorizationStatus: AuthorizationStatus.NO_AUTH,
  authInfo: {},
};

const ActionTypes = {
  REQUIRED_AUTHORIZATION: `REQUIRED_AUTHORIZATION`,
  SET_INFO: `SET_INFO`,
};

const ActionCreator = {
  requireAuthorization: (status) => {
    return {
      type: ActionTypes.REQUIRED_AUTHORIZATION,
      payload: status,
    };
  },

  setUser: (data) => {
    return {
      type: ActionTypes.SET_INFO,
      payload: data,
    };
  }
};

const reducer = (state = initialState, action) => {
  switch (action.type) {
    case ActionTypes.REQUIRED_AUTHORIZATION:
      return extend(state, {
        authorizationStatus: action.payload,
      });

    case ActionTypes.SET_INFO:
      return extend(state, {
        authInfo: action.payload,
      });
  }

  return state;
};

const Operation = {
  checkAuth: () => (dispatch, getState, api) => {
    return api.get(`/login`)
      .then((response) => {
        dispatch(ActionCreator.requireAuthorization(AuthorizationStatus.AUTH));
        dispatch(ActionCreator.setUser(EditAuthInfo.parseUser(response.data)));
      })
      .catch((err) => {
        throw err;
      });
  },

  login: (authData) => (dispatch, getState, api) => {
    return api.post(`/login`, {
      email: authData.login,
      password: authData.password,
    })
      .then((response) => {
        const user = EditAuthInfo.parseUser(response.data);
        dispatch(ActionCreator.requireAuthorization(AuthorizationStatus.AUTH));
        dispatch(ActionCreator.setUser(user));
        history.push(AppRoute.ROOT);
      });
  }
};

export {
  ActionCreator,
  ActionTypes,
  AuthorizationStatus,
  Operation,
  reducer,
};
