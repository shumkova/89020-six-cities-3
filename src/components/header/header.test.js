import React from "react";
import renderer from "react-test-renderer";
import Header from "./header";
import {AuthorizationStatus} from "../../reducer/user/user";

const userData = {
  avatar: `img/1.png`,
  email: `Oliver.conner@gmail.com`,
  id: 1,
  isPro: false,
  name: `Oliver.conner`
};

describe(`Header should render properly`, () => {
  it(`Header with Log In`, () => {
    const tree = renderer.create(
        <Header
          authorizationStatus={AuthorizationStatus.NO_AUTH}
          userData={userData}
        />
    ).toJSON;

    expect(tree).toMatchSnapshot();
  });

  it(`Header with email`, () => {
    const tree = renderer.create(
        <Header
          authorizationStatus={AuthorizationStatus.AUTH}
          userData={userData}
        />
    ).toJSON;

    expect(tree).toMatchSnapshot();
  });
});
