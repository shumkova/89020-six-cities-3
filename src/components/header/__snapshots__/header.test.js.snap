// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header should render properly Header with Log In 1`] = `[Function]`;

exports[`Header should render properly Header with email 1`] = `[Function]`;
